beacon-chain/core/epoch/epoch_processing.go

```diff
@@ -33,42 +33,76 @@ import (
-//	     # Order by the sequence of activation_eligibility_epoch setting and then index
-//	 ], key=lambda index: (state.validators[index].activation_eligibility_epoch, index))
-//	 # Dequeued validators for activation up to churn limit
-//	 for index in activation_queue[:get_validator_churn_limit(state)]:
-//	     validator = state.validators[index]
-//	     validator.activation_epoch = compute_activation_exit_epoch(get_current_epoch(state))
+// # Queue validators eligible for activation and not yet dequeued for activation
+// activation_queue = sorted([
+//
+//	index for index, validator in enumerate(state.validators)
+//	if is_eligible_for_activation(state, validator)
+//	# Order by the sequence of activation_eligibility_epoch setting and then index
+//
+// ], key=lambda index: (state.validators[index].activation_eligibility_epoch, index))
+// # Dequeued validators for activation up to churn limit
+// for index in activation_queue[:get_validator_churn_limit(state)]:
+//
+//	validator = state.validators[index]
+//	validator.activation_epoch = compute_activation_exit_epoch(get_current_epoch(state))
 func ProcessRegistryUpdates(ctx context.Context, st state.BeaconState) (state.BeaconState, error) {
-	currentEpoch := time.CurrentEpoch(st)
 	var err error
-	ejectionBal := params.BeaconConfig().EjectionBalance
+
+	previousEpoch := time.PrevEpoch(st)
+	currentEpoch := time.CurrentEpoch(st)
+	finalizedEpoch := st.FinalizedCheckpointEpoch()
+
+	isInInactivityLeak := helpers.IsInInactivityLeak(previousEpoch, finalizedEpoch)
+
+	inactivityPenaltyRate := params.BeaconConfig().InactivityPenaltyRate
+	inactivityPenaltyRatePrecision := params.BeaconConfig().InactivityPenaltyRatePrecision
+	inactivityLeakBailoutScoreThreshold := params.BeaconConfig().InactivityLeakBailoutScoreThreshold
 
 	// To avoid copying the state validator set via st.Validators(), we will perform a read only pass
 	// over the validator set while collecting validator indices where the validator copy is actually
 	// necessary, then we will process these operations.
 	eligibleForActivationQ := make([]primitives.ValidatorIndex, 0)
 	eligibleForActivation := make([]primitives.ValidatorIndex, 0)
-	eligibleForEjection := make([]primitives.ValidatorIndex, 0)
+	eligibleForBailout := make([]primitives.ValidatorIndex, 0)
 
 	if err := st.ReadFromEveryValidator(func(idx int, val state.ReadOnlyValidator) error {
 		// Collect validators eligible to enter the activation queue.
 		if helpers.IsEligibleForActivationQueue(val, currentEpoch) {
 			eligibleForActivationQ = append(eligibleForActivationQ, primitives.ValidatorIndex(idx))
 		}
 
-		// Collect validators to eject.
+		// Collect validators to bailout.
 		isActive := helpers.IsActiveValidatorUsingTrie(val, currentEpoch)
-		belowEjectionBalance := val.EffectiveBalance() <= ejectionBal
-		if isActive && belowEjectionBalance {
-			eligibleForEjection = append(eligibleForEjection, primitives.ValidatorIndex(idx))
+		pb := val.PrincipalBalance()
```

Comment:
How about making this part as a helper function like IsBelowBailoutThreshold in helpers package? Because it is also used in `beacon-chain/core/electra/registry_updates.go : ProcessRegistryUpdates()`.